{{cookiecutter.repo_name}}/.github/workflows/django_ci.yml

```diff
@@ -0,0 +1,151 @@
+name:  Django CI
+on:
+  pull_request
+jobs:
+  # uncomment when using docker
+  # docker:
+  #   name: Build the docker image
+  #   runs-on: ubuntu-latest
+  #   steps:
+  #    - uses: actions/checkout@v2
+  #    - name: Cache Docker layers
+  #      uses: actions/cache@v2
+  #      with:
+  #          path: /tmp/.buildx-cache
+  #          key: ${{ runner.os }}-buildx-${{ github.sha }}
+  #          restore-keys: |
+  #              ${{ runner.os }}-buildx-
+  #    - uses: docker/setup-buildx-action@v2
+  #      name: Set up Docker Buildx
+  #      id: buildx
+
+  #    - name: Build Docker image
+  #      uses: docker/build-push-action@v2
+  #      with:
+  #         context: ./
+  #         file: ./Dockerfile
+  #         builder: ${{ steps.buildx.outputs.name }}
+  #         cache-from: type=local,src=/tmp/.buildx-cache
+  #         cache-to: type=local,dest=/tmp/.buildx-cache
+  #         target: runtime
+
+  precommit:
+    name: Precommit linting
+    runs-on: ubuntu-latest
+    steps:
+     - uses: actions/checkout@v2
+     - uses: actions/setup-node@v3
+       with:
+           node-version: lts/*
+           cache: 'npm'
+           cache-dependency-path: '**/package-lock.json'
+
+     - run: npm install
+
+     - run: python -m pip install pre-commit
+       shell: bash
+     - uses: actions/cache@v3
+       with:
+         path: ~/.cache/pre-commit
+         key: pre-commit-3|${{ env.pythonLocation }}|${{ hashFiles('.pre-commit-config.yaml') }}
+     - run: pre-commit run --show-diff-on-failure --color=always ${{ inputs.extra_args }} --all-files
+       shell: bash
+
+
+  test:
+    name: Django CI
+    runs-on: ubuntu-latest
+    env:
+        DATABASE_URL: "psql://postgres:postgres@localhost/postgres"
+        DJANGO_SETTINGS_MODULE: "{{cookiecutter.repo_name}}.config.settings.test"
+        DJANGO_SECRET_KEY: "!!!! Change me !!!!"
+        ACUITY_USER_KEY: ""
+        ACUITY_SECRET_KEY: ""
+
+    steps:
+     - uses: actions/checkout@v2
+
+     - uses: actions/setup-node@v3
+       with:
+           node-version: lts/*
+           cache: 'npm'
+           cache-dependency-path: '**/package-lock.json'
+     - run: npm install
+     - run: npm run build
+
+     - name: Load cached Poetry installation
+       uses: actions/cache@v2
+       with:
+           path: ~/.local
+           key: poetry-0  # increment to reset cache
+
+     - name: Install and configure Poetry
+       uses: snok/install-poetry@v1
+       with:
+           version: 1.2.0b1
+           virtualenvs-create: true
+           virtualenvs-in-project: true
+           installer-parallel: true
+
+     - uses: actions/setup-python@v4
+       with:
+           python-version-file: '.python-version'
+           cache: 'poetry'
+
+     - run: poetry env use 3.10.2 #TODO: get this dynamic on python version
```

Comment:
This can be done by using `setup-python` output.

```
      - uses: actions/setup-python@v4
        id: setup_python
        with:
          python-version-file: ".python-version"
          cache: "poetry"

      - run: poetry env use ${{ steps.setup_python.outputs.python-version }}
```